mulog-prometheus/src/com/brunobonacci/mulog/publishers/prometheus.clj

```diff
@@ -41,31 +49,99 @@
   {:max-items     1000
    :publish-delay 100
-   ;; publish-method :push-gateway | :scrape
-   ;; push-gateway-endpoint ""
-   ;; :register    (...)
-   ;; ....
-
-   ;; function to transform records
-   :transform     identity
-   })
-
 
+   ;; You can supply your own registry which will be used for all events.
+   ;; If you do not specify a registry, the default registry is used.
+   ;; The default registry is the static `CollectorRegistry.defaultRegistry`
+   :registry      (reg/create-default)
+
+   ;; You can setup the prometheus-publisher to push to a prometheus push gateway.
+   ;; `job` is always required - These are what your metrics are identified by.
+   ;; Typically a unique job name or application name.
+   ;;
+   ;; You can then either supply your own `^PushGateway :gateway` or supply an
+   ;; `:endpoint` which will create a new `PushGateway`.
+   :push-gateway  {:endpoint nil
+                   :job      nil
+                   :gateway  nil}
+
+   ;; A function to apply to the sequence of events before publishing.
+   ;; This transformation function can be used to filter, transform,
+   ;; anonymise events before they are published to a external system.
+   ;; by defatult there is no transformation.  (since v0.1.8)
+   :transform         identity
+
+   ;; A function to apply to the sequence of metrics before converting into a collection.
+   ;; You will receive a `com.brunobonacci.mulog.publishers.prometheus.metrics-spec/metric` and
+   ;; should return the same or nil.
+   ;; This tranformation function can be used to remove metric types, (alter/add/remove) labels
+   ;; or add more detailed descriptions of what the metric does.
+   ;; 
+   ;; For the :histogram metric type. You can supply buckets which will be used instead of the defaults.
+   ;; default: [0.005 0.01 0.025 0.05 0.075 0.1 0.25 0.5 0.75 1 2.5 5 7.5 10]
+   ;; 
+   ;; For the :summary metric type. You can suppy the following to be used instead of the defaults.
+   ;; quantiles - quantiles to be used over the sliding window of time.
+   ;; default: [[0.5 0.001][0.9 0.001][0.95 0.001][0.99 0.001][0.999 0.001]]
+   ;; max-age-seconds - duration of the time window.
+   ;; default: 600
+   ;; age-buckets - buckets used to implement sliding window.
+   ;; default: 5
+   ;; 
+   ;; by default there is no transformation.
+   :transform-metrics identity})
 
 (defn prometheus-publisher
   [config]
-  (PrometheusPublisher.
-    (merge DEFAULT-CONFIG config)
-    (rb/agent-buffer 10000)
-    (or (:transform config) identity)))
+  (let [{{:keys [endpoint job gateway]} :push-gateway :as config} (merge DEFAULT-CONFIG config)]
+    (assoc-in config [:push-gateway :gateway]
+      (when (and (not gateway) endpoint job)
+        (PushGateway. (as-url endpoint))))
+    ;; create the prometheus publisher
+    (PrometheusPublisher.
+      config
+      (rb/agent-buffer 10000)
+      (or (:registry config) (reg/create-default))
+      (or (:transform :registry) identity))))
```

Comment:
`(:transform :registry)` this will always return `nil`

Reply:
Good spot, typo during refactor thanks